include/cooking/cookingJavascript/searchbar.php

```diff
@@ -0,0 +1,20 @@
+<script>
+function searchBar() {
+    // Declare variables
+    var input, filter, ul, li, a, i;
+    input = document.getElementById('myInput');
+    filter = input.value.toUpperCase();
+    ul = document.getElementById("myUL");
+    li = myUL.getElementsByTagName('li');
+
+    // Loop through all list items, and hide those who don't match the search query
+    for (i = 0; i < li.length; i++) {
+        a = li[i].getElementsByTagName("a")[0];
+        if (a.innerHTML.toUpperCase().indexOf(filter) > -1) {
+            li[i].style.display = "";
+        } else {
+            li[i].style.display = "none";
+        }
+    }
+}
```

Comment:
You may know this already, but I just want to make sure: client-side search like this is great from a user experience standpoint, but it can be a bit limiting because you might not always have all the data in the client. I'm not actually sure what you're searching here, but if it's a list of recipes or something like that, it wouldn't work on big recipe lists. Like, if you have 100,000 recipes in your db, you wouldn't want to load them all in the client, so you couldn't search them with js. In that case, you'd want to do server-side search.